src/main/java/duke/Task.java

```diff
@@ -0,0 +1,59 @@
+package duke;
+
+public class Task {
+    protected String description;
+    protected boolean isDone;
+    protected int index;
+
+    /**
+     * Constructor for the Task class
+     * @param index the index of the task
+     * @param description the description of the task
+     */
+    public Task(int index, String description) {
+        this.index = index + 1;
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * marks the task as done
+     */
+    public void mark() {
+        this.isDone = true;
+    }
+
+    /**
+     * marks the task as not done
+     */
+    public void unmark() {
+        this.isDone = false;
+    }
+
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
```

Comment:
Perhaps this is a more acceptable way to format a ternary expression?
```suggestion
        return (isDone) ? "X" : " "; // mark done task with X
```